aicsimageio/readers/czi_reader.py

```diff
@@ -937,16 +948,74 @@ def get_mosaic_tile_position(self, mosaic_tile_index: int) -> Tuple[int, int]:
         ------
         UnexpectedShapeError
             The image has no mosaic dimension available.
-        IndexError
-            No matching mosaic tile index found.
+
+        Notes
+        -----
+        Defaults T and C dimensions to 0 if present as dimensions in image
+        to avoid reading in massive image stack for large files.
         """
         if DimensionNames.MosaicTile not in self.dims.order:
             raise exceptions.UnexpectedShapeError("No mosaic dimension in image.")
 
-        # Get max of mosaic positions from lif
         with self._fs.open(self._path) as open_resource:
             czi = CziFile(open_resource.f)
 
-            bboxes = czi.get_all_mosaic_tile_bounding_boxes(S=self.current_scene_index)
-            bbox = list(bboxes.values())[mosaic_tile_index]
+            # Default T and C dimensions to 0 to improve
+            # worst case read time for large files **only**
+            # when those dimensions are present on the image.
+            if "T" not in kwargs and "T" in self.dims.order:
+                kwargs["T"] = 0
+            if "C" not in kwargs and "C" in self.dims.order:
+                kwargs["C"] = 0
+
+            bbox = czi.get_mosaic_tile_bounding_box(
+                M=mosaic_tile_index, S=self.current_scene_index, **kwargs
+            )
             return bbox.y, bbox.x
+
+    def get_mosaic_tile_positions(self, **kwargs: int) -> List[Tuple[int, int]]:
+        """
+        Get the absolute positions of the top left points for each mosaic tile
+        matching the specified dimensions and current scene.
+
+        Parameters
+        ----------
+        kwargs: int
+            The keywords below allow you to specify the dimensions that you wish
+            to match. If you under-specify the constraints you can easily
+            end up with a massive image stack.
+                       Z = 1   # The Z-dimension.
+                       C = 2   # The C-dimension ("channel").
+                       T = 3   # The T-dimension ("time").
+
+        Returns
+        -------
+        mosaic_tile_positions: List[Tuple[int, int]]
+            List of the Y and X coordinate for the tile positions.
+
+        Raises
+        ------
+        UnexpectedShapeError
+            The image has no mosaic dimension available.
+        """
+        if DimensionNames.MosaicTile not in self.dims.order:
```

Comment:
This is used here, but `"T"` and `"C"` are used directly above - should that be changed so the latter is consistent?

Reply:
Good catch, I agree we should use our internal symbolic constant names for these even though they are so much more verbose.

Reply:
Great idea, will do!